### ISSUE TITLE
nameserver discovery

### ISSUE THREAD

**Issue by DeTeam** [shuhei/pollen#19]
One thing that pollen could do is to find nameservers to query based on dns configuration, `/etc/resolv.conf`, this way one would not have to pass it manually.

Could also provide it as a utility function

**Comment by shuhei** [comment]
Oh, I thought that `dns.resolve4()` used nameservers in `/etc/resolv.conf` because [the underlying c-ares seems to do so](https://c-ares.haxx.se/ares_init_options.html). Have you tested whether it works like that?

**Comment by shuhei** [comment]
In the worst case, we could use `dns.lookup()` with `all: true` instead of `dns.resolve4()` so that we can support `/etc/hosts` too. There will be less chance that it will block the libuv thread pool as long as we don't make too many lookups.

**Comment by DeTeam** [comment]
> There will be less chance that it will block the libuv thread pool as long as we don't make too many lookups.

Btw, I was reading libuv source code the other day, dns request do not seem to block the whole pool.
[Only half](https://github.com/libuv/libuv/blob/v1.x/src/threadpool.c#L45-L47) of uv thread pool is dedicated to slow io (dns), the rest is available for fast io (filesystem) and cpu-heavy tasks.

>  I thought that `dns.resolve4()` used nameservers in

Ah, you're right, I got confused then. Closing the ticket!

> Have you tested whether it works like that?

Nope, I'm just browsing through the source atm
